extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/filters/accesslog/AccessLogHandler.java

```diff
@@ -145,7 +151,21 @@ public void handle(RoutingContext rc) {
         QuarkusRequestWrapper.get(rc.request()).addRequestDoneHandler(new Handler<Void>() {
             @Override
             public void handle(Void event) {
+                Map<String, Optional<String>> serialized = ((ExchangeAttributeSerializable) tokens).serialize(rc);
+                Set<String> keys = new HashSet<>();
+
+                for (Map.Entry<String, Optional<String>> entry : serialized.entrySet()) {
+                    if (entry.getValue().isPresent()) {
+                        keys.add(entry.getKey());
+                        MDC.put(entry.getKey(), entry.getValue().get());
+                    }
+                }
+
                 accessLogReceiver.logMessage(tokens.readAttribute(rc));
+
+                for (String key : keys) {
+                    MDC.remove(key);
+                }
```

Comment:
I still believe that this should restore the previous values of the keys rather than clearing them.